features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreMapFragment.kt

```diff
@@ -192,82 +177,98 @@ class ExploreMapFragment: SupportMapFragment() {
 
     private fun addMarkerOnCurrentPosition() {
         val bitmap = getBitmapFromDrawable(R.drawable.user_location_map_marker)
-        currentLocationMarker = markerCollection?.addMarker(MarkerOptions().apply {
+        currentLocationMarker = googleMap?.addMarker(MarkerOptions().apply {
             position(mCurrentUserLocation)
             anchor(0.5f, 0.5f)
             if (bitmap != null) {
                 icon(BitmapDescriptorFactory.fromBitmap(bitmap))
             }
             draggable(true)
             zIndex(5f)
-        }).apply {
-            this?.tag = CURRENT_POSITION_MARKER_TAG
-        }
+        })
     }
 
     private fun showLocationOnMap() {
         Log.e(this::class.java.simpleName, "Lat: ${mCurrentUserLocation.latitude}, Lng: ${mCurrentUserLocation.longitude}")
-        if (currentLocationCircle != null) circleCollection?.remove(currentLocationCircle)
-        if (currentLocationMarker != null) markerCollection?.remove(currentLocationMarker)
 
-        addMarkerOnCurrentPosition()
-        addCircleAroundCurrentPosition()
-        setMapDefaultViewLevel(viewModel.radius)
+        if (currentLocationCircle == null) {
+            addCircleAroundCurrentPosition()
+        }
+
+        if (currentLocationMarker == null) {
+            addMarkerOnCurrentPosition()
+        }
+
+        currentLocationMarker?.position = mCurrentUserLocation
+        currentLocationCircle?.center = mCurrentUserLocation
+        currentLocationCircle?.radius = currentAccuracy
```

Comment:
Instead of re-creating the current location marker and circle, we can simply update their locations.